facade/service.go

```diff
@@ -1213,18 +1213,86 @@ func (f *Facade) scheduleService(ctx datastore.Context, tenantID, serviceID stri
 		return 0, err
 	}
 
-	var affected int
+	serviceScheduler := func() (int, error) {
+		affected := 0
+		var errToReturn error = nil
+		if emergency {
+			// Sort the services by EmergencyShutdownLevel - 1
+			sort.Sort(service.ByEmergencyShutdown{svcs})
+
+			// Start one group at a time
+			if len(svcs) > 0 {
+				previousLevel := svcs[0].EmergencyShutdownLevel
+				nextBatch := []*service.Service{}
+				nextBatchIDs := []string{}
+				for _, svc := range svcs {
+					// Set EmergencyShutdown to true and update the database
+					svc.EmergencyShutdown = true
+					uerr := f.updateService(ctx, tenantID, *svc, false, false)
+					if uerr != nil {
+						errToReturn = uerr
+						logger.WithField("service", svc.ID).WithError(uerr).Error("Failed to update database with EmergencyShutdown")
+					}
+
+					currentLevel := svc.EmergencyShutdownLevel
+					if currentLevel == previousLevel {
+						nextBatch = append(nextBatch, svc)
+						nextBatchIDs = append(nextBatchIDs, svc.ID)
+					} else {
+						// Schedule this batch
+						levelLogger := logger.WithField("level", previousLevel)
+						levelLogger.Info("Shutting down all services at current emergency shutdown level")
+						a, serr := scheduleServices(f, nextBatch, ctx, tenantID, serviceID, desiredState)
```

Comment:
The services in the batch with ES level 0 need to be ordered (and batched) by reverse StartLevel.